docker.go

```diff
@@ -1145,9 +1187,14 @@ func (p *DockerProvider) ReuseOrCreateContainer(ctx context.Context, req Contain
 		terminationSignal: termSignal,
 		stopProducer:      nil,
 		logger:            p.Logger,
-		isRunning:         c.State == "running",
 	}
 
+	if err := waitContainerReadyWithLog(ctx, dc); err != nil {
```

Comment:
Why do we need to wait for the container here? The code should have already waited for it the first time it was created, right?

Reply:
Not really. Waiting will only work correctly for the first caller. The other callers, if they start working before the first caller finishes waiting, will consider the container to be completely ready, although it is not. waiting in this place fixes this problem.

You can check this by removing the `waitContainerReadyWithLog` function call (and the log check in the test) and run the `TestGenericReusableContainerInSubprocess` test